sdk/src/main/java/io/dapr/actors/DaprHttpAsyncClient.java

```diff
@@ -0,0 +1,186 @@
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<Void> unregisterReminder(String actorType, String actorId, String reminderName) {
+        String url = String.format(Constants.ACTOR_REMINDER_RELATIVE_URL_FORMAT, actorType, actorId, reminderName);
+        return invokeAPIVoid("DELETE", url, null);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<Void> registerTimer(String actorType, String actorId, String timerName, String data) {
+        String url = String.format(Constants.ACTOR_TIMER_RELATIVE_URL_FORMAT, actorType, actorId, timerName);
+        return invokeAPIVoid("PUT", url, data);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<Void> unregisterTimerAsync(String actorType, String actorId, String timerName) {
+        String url = String.format(Constants.ACTOR_TIMER_RELATIVE_URL_FORMAT, actorType, actorId, timerName);
+        return invokeAPIVoid("DELETE", url, null);
+    }
+
+    /**
+     * Invokes an API asynchronously that returns Void.
+     * @param method HTTP method.
+     * @param urlString url as String.
+     * @param json JSON payload or null.
+     * @return Asynchronous Void
+     */
+    private final Mono<Void> invokeAPIVoid(String method, String urlString, String json) {
+        return this.invokeAPI(method, urlString, json).then();
+    }
+
+    /**
+     * Invokes an API asynchronously that returns a text payload.
+     * @param method HTTP method.
+     * @param urlString url as String.
+     * @param json JSON payload or null.
+     * @return Asynchronous text
+     */
+    private final Mono<String> invokeAPI(String method, String urlString, String json) {
+        return Mono.fromSupplier(() -> {
+            try {
+                return tryInvokeAPI(method, urlString, json);
+            } catch (IOException e) {
+                e.printStackTrace();
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    /**
+     * Invokes an API synchronously and returns a text payload.
+     * @param method HTTP method.
+     * @param urlString url as String.
+     * @param json JSON payload or null.
+     * @return text
+     */
+    private final String tryInvokeAPI(String method, String urlString, String json) throws IOException {
+        String requestId = UUID.randomUUID().toString();
+        RequestBody body = json != null ? RequestBody.create(MEDIA_TYPE_APPLICATION_JSON, json) : REQUEST_BODY_EMPTY_JSON;
+
+        Request request = new Request.Builder()
+                .url(new URL(this.baseUrl + urlString))
+                .method(method, body)
+                .addHeader(Constants.HEADER_DAPR_REQUEST_ID, requestId)
+                .build();
+
+        Response response = this.httpClient.newCall(request).execute();
+        if (!response.isSuccessful())
+        {
+            throw new IOException(String.format("Dapr's Actor API %s failed with return code: %d.", urlString, response.code()));

```

Comment:
In the failure case, Dapr returns a json with error code and an error messgae, the code can try to parse and throw a more specific DaprEcception wimilar to https://github.com/dapr/dotnet-sdk/blob/master/src/Dapr.Actors/DaprHttpInteractor.cs#L446

Reply:
I did parse the error code. I am doing a simpler error handling for now.

We will enhance once more use cases are implemented. I found the error handling in the .Net SDK a little bit complicated. Plus, the parsing from String to Enum seems to be overkill since the SDK will probably never keep up with the changes on the server side.

Reply:
Sure, ErrorCode can remain as string instead of enum. but in general SDK should keep up with the runtime